### PR TITLE
Feature/list object v2 with metadata

### PR DESCRIPTION
**Pull request by Max9419**
Hello!

So this is a PR for [this requested feature](https://github.com/minio/minio-js/issues/822)!

I've isolated the new feature in an extension class, like asked.

I think the way I did it is the best way but if you have any suggestions, feel free to share!

I've also added an example and the functional test to mirror the rest of the project.

Thank you!

PS : There is no documentation for the endpoint on https://docs.minio.io/. I'm not sure what I should do here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,7 @@ The full API Reference is available here.
 * [`removeBucket`](https://docs.min.io/docs/javascript-client-api-reference#removeBucket)
 * [`listObjects`](https://docs.min.io/docs/javascript-client-api-reference#listObjects)
 * [`listObjectsV2`](https://docs.min.io/docs/javascript-client-api-reference#listObjectsV2)
+* `listObjectsV2WithMetadata` (Extension)
```
Comment (harshavardhana) [inline README.md]: The doc reference should be similar to `listObjectsV2` with a hyperlink, relevant docs should be added in docs/API.md

File: src/main/extensions.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import Stream from 'stream'
```
Comment (harshavardhana) [inline src/main/extensions.js]: missing copyright of this project

Comment (Max9419) [reply src/main/extensions.js]: Done

File: README.md (hunk 1 above)
Comment (Max9419) [reply README.md]: This should do it

File: src/main/extensions.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
+/*
+ * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2015 MinIO, Inc.
```
Comment (kannappanr) [inline src/main/extensions.js]:
```suggestion
 * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```

File: examples/list-objects-v2-with-metadata.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
+ * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2015 MinIO, Inc.
```
Comment (kannappanr) [inline examples/list-objects-v2-with-metadata.js]:
```suggestion
 * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```

Comment (Max9419) [reply examples/list-objects-v2-with-metadata.js]: my bad, thank you

**Pull request merged**